package/MDAnalysis/analysis/nuclinfo.py

```diff
@@ -336,24 +364,45 @@ def tors(universe, seg, i):
-                             " atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i))
-    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i), " atom {0!s} {1!s} C3\' ".format(seg, i),
-                             " atom {0!s} {1!s} O3\' ".format(seg, i), " atom {0!s} {1!s} P    ".format(seg, i + 1))
-    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i),
-                             " atom {0!s} {1!s} P    ".format(seg, i + 1), " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
+    a = universe.select_atoms(" atom {0!s} {1!s} O3\' ".format(seg, i - 1),
+                              " atom {0!s} {1!s} P  ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i))
+
+    b = universe.select_atoms(" atom {0!s} {1!s} P    ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i))
+
+    g = universe.select_atoms(" atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i))
+
+    d = universe.select_atoms(" atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i))
+
+    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1))
+
+    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
     try:
-        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i), " atom {0!s} {1!s} C1\' ".format(seg, i),
-                                 " atom {0!s} {1!s} N1 ".format(seg, i), " atom {0!s} {1!s} C2  ".format(seg, i))
+        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i),
+                                  " atom {0!s} {1!s} C1\' ".format(seg, i),
+                                  " atom {0!s} {1!s} N1 ".format(seg, i),
+                                  " atom {0!s} {1!s} C2  ".format(seg, i))
     except:
```

Comment:
There should probably be an issue on the selection issue then.

On 19-01-16 19:26, Richard Gowers wrote:
>
> In package/MDAnalysis/analysis/nuclinfo.py 
> <https://github.com/MDAnalysis/mdanalysis/pull/640#discussion_r50152210>:
>
> >      try:
> > -        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i), " atom {0!s} {1!s} C1\' ".format(seg, i),
> > -                                 " atom {0!s} {1!s} N1 ".format(seg, i), " atom {0!s} {1!s} C2  ".format(seg, i))
> > +        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i),
> > +                                  " atom {0!s} {1!s} C1\' ".format(seg, i),
> > +                                  " atom {0!s} {1!s} N1 ".format(seg, i),
> > +                                  " atom {0!s} {1!s} C2  ".format(seg, i))
> >      except:
>
> Yep, but then the tests will fail because that will raise an 
> IndexError. I think this block was hiding a bug in the selection system
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/MDAnalysis/mdanalysis/pull/640/files#r50152210>.
>